dbms/src/Functions/bitBoolMaskAnd.cpp

```diff
@@ -0,0 +1,35 @@
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionBinaryArithmetic.h>
+#include <DataTypes/NumberTraits.h>
+
+namespace DB
+{
+
+    template <typename A, typename B>
+    struct BitBoolMaskAndImpl
```

Comment:
Missing comments. I spent about a minute to figure out that this function does bit and of least significant bit and bit or of second bit.